tests/databases/test_redshift.py

```diff
@@ -0,0 +1,324 @@
+    "database_table_fixture",
+    [
+        {"database": Database.REDSHIFT},
+    ],
+    indirect=True,
+    ids=["redshift"],
+)
+def test_load_pandas_dataframe_to_table(database_table_fixture):
+    """Test load_pandas_dataframe_to_table against redshift"""
+    database, table = database_table_fixture
+
+    pandas_dataframe = pd.DataFrame(data={"id": [1, 2]})
+    database.load_pandas_dataframe_to_table(pandas_dataframe, table)
+
+    statement = f"SELECT * FROM {database.get_table_qualified_name(table)};"
+    response = database.run_sql(statement)
+
+    rows = response.fetchall()
+    assert len(rows) == 2
+    assert rows[0] == (1,)
+    assert rows[1] == (2,)
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {"database": Database.REDSHIFT},
+    ],
+    indirect=True,
+    ids=["redshift"],
+)
+def test_load_file_to_table(database_table_fixture):
+    """Test loading on files to redshift database"""
+    database, target_table = database_table_fixture
+    filepath = str(pathlib.Path(CWD.parent, "data/sample.csv"))
+    database.load_file_to_table(File(filepath), target_table, {})
+
+    df = database.hook.get_pandas_df(
+        f"SELECT * FROM {database.get_table_qualified_name(target_table)}"
+    )
+    assert len(df) == 3
+    expected = pd.DataFrame(
+        [
+            {"id": 1, "name": "First"},
+            {"id": 2, "name": "Second"},
+            {"id": 3, "name": "Third with unicode पांचाल"},
+        ]
+    )
+    test_utils.assert_dataframes_are_equal(df, expected)
+
+
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {"database": Database.REDSHIFT},
+    ],
+    indirect=True,
+    ids=["redshift"],
+)
+def test_export_table_to_file_file_already_exists_raises_exception(
+    database_table_fixture,
+):
+    """
+    Test export_table_to_file_file() where the end file already exists, should result in exception
+    when the override option is False
+    """
+    database, source_table = database_table_fixture
+    filepath = pathlib.Path(CWD.parent, "data/sample.csv")
+    with pytest.raises(FileExistsError) as exception_info:
+        database.export_table_to_file(source_table, File(str(filepath)))
+    err_msg = exception_info.value.args[0]
+    assert err_msg.endswith(f"The file {filepath} already exists.")
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [
+        {
+            "database": Database.REDSHIFT,
+            "file": File(str(pathlib.Path(CWD.parent, "data/sample.csv"))),
+        },
+    ],
+    indirect=True,
+    ids=["redshift"],
+)
+def test_export_table_to_file_overrides_existing_file(database_table_fixture):
+    """
+    Test export_table_to_file_file() where the end file already exists,
+    should result in overriding the existing file
+    """
+    database, populated_table = database_table_fixture
+
+    filepath = str(pathlib.Path(CWD.parent, "data/sample.csv"))
```

Comment:
@pankajkoti How do we know the file is replaced? I see the same file in the fixture and test getting loaded.

Reply:
Right. Creating a new temporary file with different initial values and then overriding the file.